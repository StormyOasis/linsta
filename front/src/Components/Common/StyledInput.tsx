import React from "react";
import { styled } from "styled-components";

import * as styles from "./Common.module.css";

const StyledInputWrapper = styled.div`
  display: flex;
  flex-direction: row;
  margin: 0 40px 5px 40px;
`;

const StyledInputInput = styled.input<{"data-testid"?: string}>`
  border: 2px solid ${props => props.theme['colors'].borderDefaultColor};
  background-color: ${props => props.theme['colors'].inputBackgroundColor};
  display: block;
  width: 100%;
  height: 24px;
`;

const StyledInputLabel = styled.label`
    position: relative;
    width: 100%;
`;

type StyledInputProps = {
  name: string;
  value?: string | Number;
  placeholder?: string;
  isValid?: boolean;
  onChange?: any;
  type?: string;
  maxLength?: number;
  style?: any;
  datatestid?: string;
}

const StyledInput: React.FC<StyledInputProps> = (props: StyledInputProps) => {

    // Determine validation classname (None, valid, invalid)
    const value = `${props.value}`;
    let validationClass = "";
    if(value.length != 0) {
      validationClass = props.isValid ? styles.validationPass : styles.validationFail;
    }

    return (
      <StyledInputWrapper>
        <StyledInputLabel className={validationClass}>
            <StyledInputInput
                style={props.style}
                type={props.type}
                name={props.name}
                value={value}
                placeholder={props.placeholder}
                onChange={props.onChange} 
                maxLength={props.maxLength ? props.maxLength : 255}
                data-testid={props.datatestid}
                />
        </StyledInputLabel>
      </StyledInputWrapper>        
    );
}

export default StyledInput;