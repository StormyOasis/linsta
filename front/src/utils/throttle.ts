import { useRef } from 'react';

const useThrottle = (callback: Function, delay: number) => {
    const lastCall = useRef(0);

    return (...args: any[]) => {
        const now = new Date().getTime();
        if (now - lastCall.current >= delay) {
            lastCall.current = now;
            callback(...args);
        }
    };
};

export default useThrottle;