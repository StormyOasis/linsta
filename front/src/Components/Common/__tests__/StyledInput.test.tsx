import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import StyledInput from "../StyledInput";
import '@testing-library/jest-dom';
import { renderWithStore } from '../../../utils/test-utils';


// Mock icons
jest.mock("../Icon", () => ({
    CheckCircleSVG: (props: any) => <svg data-testid="check-circle-svg" {...props} />,
    CircleXSVG: (props: any) => <svg data-testid="circle-x-svg" {...props} />,
}));

describe("StyledInput", () => {
    it("renders input with correct props", () => {
        const r = renderWithStore(
            <StyledInput
                name="test"
                value="hello"
                placeholder="Type here"
                onChange={jest.fn()}
                shouldValidate={false}
                datatestid="styled-input"
            />
        );
        const input = screen.getByTestId("styled-input");
        expect(input).toBeInTheDocument();
        expect(input).toHaveAttribute("name", "test");
        expect(input).toHaveAttribute("placeholder", "Type here");
        expect(input).toHaveValue("hello");
        expect(r).toMatchSnapshot();
    });

    it("calls onChange when input changes", () => {
        const handleChange = jest.fn();
        const r = renderWithStore(
            <StyledInput
                value=""
                onChange={handleChange}
                shouldValidate={false}
                datatestid="styled-input"
            />
        );
        const input = screen.getByTestId("styled-input");
        fireEvent.change(input, { target: { value: "abc" } });
        expect(handleChange).toHaveBeenCalled();
        expect(r).toMatchSnapshot();
    });

    it("shows CheckCircleSVG when valid and shouldValidate is true", () => {
        const r = renderWithStore(
            <StyledInput
                value="abc"
                isValid={true}
                shouldValidate={true}
                datatestid="styled-input"
            />
        );
        expect(screen.getByTestId("check-circle-svg")).toBeInTheDocument();
        expect(r).toMatchSnapshot();
    });

    it("shows CircleXSVG when invalid and shouldValidate is true", () => {
        const r = renderWithStore(
            <StyledInput
                value="abc"
                isValid={false}
                shouldValidate={true}
                datatestid="styled-input"
            />
        );
        expect(screen.getByTestId("circle-x-svg")).toBeInTheDocument();
        expect(r).toMatchSnapshot();
    });

    it("does not show validation icon if value is empty", () => {
        const r = renderWithStore(
            <StyledInput
                value=""
                isValid={false}
                shouldValidate={true}
                datatestid="styled-input"
            />
        );
        expect(screen.queryByTestId("check-circle-svg")).not.toBeInTheDocument();
        expect(screen.queryByTestId("circle-x-svg")).not.toBeInTheDocument();
        expect(r).toMatchSnapshot();
    });

    it("applies noMargin and width props", () => {
        const r = renderWithStore(
            <StyledInput
                value="abc"
                shouldValidate={false}
                noMargin={true}
                width="200px"
                datatestid="styled-input"
            />
        );
        const wrapper = screen.getByTestId("styled-input").closest("div");
        expect(wrapper).toHaveStyle("margin: 0");
        expect(wrapper).toHaveStyle("width: 200px");
        expect(r).toMatchSnapshot();
    });

    it("applies noBorder prop", () => {
        const r = renderWithStore(
            <StyledInput
                value="abc"
                shouldValidate={false}
                noBorder={true}
                datatestid="styled-input"
            />
        );
        const input = screen.getByTestId("styled-input");
        expect(input).toHaveStyle("border: none");
        expect(r).toMatchSnapshot();
    });

    it("calls onClick when input is clicked", () => {
        const handleClick = jest.fn();
        const r = renderWithStore(
            <StyledInput
                value="abc"
                shouldValidate={false}
                onClick={handleClick}
                datatestid="styled-input"
            />
        );
        const input = screen.getByTestId("styled-input");
        fireEvent.click(input);
        expect(handleClick).toHaveBeenCalled();
        expect(r).toMatchSnapshot();
    });
});