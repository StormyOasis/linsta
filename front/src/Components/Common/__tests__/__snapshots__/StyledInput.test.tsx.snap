// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledInput applies noBorder prop 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 StyledInput__StyledInputWrapper-sc-fkf055-0 escqfF ckFvxY fjNDtr cyFDIt"
      >
        <label
          class="StyledInput__StyledInputLabel-sc-fkf055-2 iHXgrc"
        >
          <input
            class="StyledInput__StyledInputInput-sc-fkf055-1 iTNlXn"
            data-testid="styled-input"
            maxlength="255"
            value="abc"
          />
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 StyledInput__StyledInputWrapper-sc-fkf055-0 escqfF ckFvxY fjNDtr cyFDIt"
    >
      <label
        class="StyledInput__StyledInputLabel-sc-fkf055-2 iHXgrc"
      >
        <input
          class="StyledInput__StyledInputInput-sc-fkf055-1 iTNlXn"
          data-testid="styled-input"
          maxlength="255"
          value="abc"
        />
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`StyledInput applies noMargin and width props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 StyledInput__StyledInputWrapper-sc-fkf055-0 lgTzGj ckFvxY fjNDtr cUrOTr"
      >
        <label
          class="StyledInput__StyledInputLabel-sc-fkf055-2 iHXgrc"
        >
          <input
            class="StyledInput__StyledInputInput-sc-fkf055-1 kkxInS"
            data-testid="styled-input"
            maxlength="255"
            value="abc"
          />
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 StyledInput__StyledInputWrapper-sc-fkf055-0 lgTzGj ckFvxY fjNDtr cUrOTr"
    >
      <label
        class="StyledInput__StyledInputLabel-sc-fkf055-2 iHXgrc"
      >
        <input
          class="StyledInput__StyledInputInput-sc-fkf055-1 kkxInS"
          data-testid="styled-input"
          maxlength="255"
          value="abc"
        />
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`StyledInput calls onChange when input changes 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 StyledInput__StyledInputWrapper-sc-fkf055-0 escqfF ckFvxY fjNDtr cyFDIt"
      >
        <label
          class="StyledInput__StyledInputLabel-sc-fkf055-2 iHXgrc"
        >
          <input
            class="StyledInput__StyledInputInput-sc-fkf055-1 kkxInS"
            data-testid="styled-input"
            maxlength="255"
            value=""
          />
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 StyledInput__StyledInputWrapper-sc-fkf055-0 escqfF ckFvxY fjNDtr cyFDIt"
    >
      <label
        class="StyledInput__StyledInputLabel-sc-fkf055-2 iHXgrc"
      >
        <input
          class="StyledInput__StyledInputInput-sc-fkf055-1 kkxInS"
          data-testid="styled-input"
          maxlength="255"
          value=""
        />
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`StyledInput calls onClick when input is clicked 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 StyledInput__StyledInputWrapper-sc-fkf055-0 escqfF ckFvxY fjNDtr cyFDIt"
      >
        <label
          class="StyledInput__StyledInputLabel-sc-fkf055-2 iHXgrc"
        >
          <input
            class="StyledInput__StyledInputInput-sc-fkf055-1 kkxInS"
            data-testid="styled-input"
            maxlength="255"
            value="abc"
          />
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 StyledInput__StyledInputWrapper-sc-fkf055-0 escqfF ckFvxY fjNDtr cyFDIt"
    >
      <label
        class="StyledInput__StyledInputLabel-sc-fkf055-2 iHXgrc"
      >
        <input
          class="StyledInput__StyledInputInput-sc-fkf055-1 kkxInS"
          data-testid="styled-input"
          maxlength="255"
          value="abc"
        />
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`StyledInput does not show validation icon if value is empty 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 StyledInput__StyledInputWrapper-sc-fkf055-0 escqfF ckFvxY fjNDtr cyFDIt"
      >
        <label
          class="StyledInput__StyledInputLabel-sc-fkf055-2 iHXgrc"
        >
          <input
            class="StyledInput__StyledInputInput-sc-fkf055-1 kkxInS"
            data-testid="styled-input"
            maxlength="255"
            value=""
          />
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 StyledInput__StyledInputWrapper-sc-fkf055-0 escqfF ckFvxY fjNDtr cyFDIt"
    >
      <label
        class="StyledInput__StyledInputLabel-sc-fkf055-2 iHXgrc"
      >
        <input
          class="StyledInput__StyledInputInput-sc-fkf055-1 kkxInS"
          data-testid="styled-input"
          maxlength="255"
          value=""
        />
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`StyledInput renders input with correct props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 StyledInput__StyledInputWrapper-sc-fkf055-0 escqfF ckFvxY fjNDtr cyFDIt"
      >
        <label
          class="StyledInput__StyledInputLabel-sc-fkf055-2 iHXgrc"
        >
          <input
            class="StyledInput__StyledInputInput-sc-fkf055-1 kkxInS"
            data-testid="styled-input"
            maxlength="255"
            name="test"
            placeholder="Type here"
            value="hello"
          />
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 StyledInput__StyledInputWrapper-sc-fkf055-0 escqfF ckFvxY fjNDtr cyFDIt"
    >
      <label
        class="StyledInput__StyledInputLabel-sc-fkf055-2 iHXgrc"
      >
        <input
          class="StyledInput__StyledInputInput-sc-fkf055-1 kkxInS"
          data-testid="styled-input"
          maxlength="255"
          name="test"
          placeholder="Type here"
          value="hello"
        />
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`StyledInput shows CheckCircleSVG when valid and shouldValidate is true 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 StyledInput__StyledInputWrapper-sc-fkf055-0 escqfF ckFvxY fjNDtr cyFDIt"
      >
        <label
          class="StyledInput__StyledInputLabel-sc-fkf055-2 iHXgrc"
        >
          <input
            class="StyledInput__StyledInputInput-sc-fkf055-1 kkxInS"
            data-testid="styled-input"
            maxlength="255"
            value="abc"
          />
          <div
            class="CombinedStyling__Div-sc-c3jdxk-0 StyledInput__ValidationIcon-sc-fkf055-3 hNEsIT kYVJfy"
          >
            <svg
              data-testid="check-circle-svg"
              fill="currentColor"
              stroke="none"
            />
          </div>
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 StyledInput__StyledInputWrapper-sc-fkf055-0 escqfF ckFvxY fjNDtr cyFDIt"
    >
      <label
        class="StyledInput__StyledInputLabel-sc-fkf055-2 iHXgrc"
      >
        <input
          class="StyledInput__StyledInputInput-sc-fkf055-1 kkxInS"
          data-testid="styled-input"
          maxlength="255"
          value="abc"
        />
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 StyledInput__ValidationIcon-sc-fkf055-3 hNEsIT kYVJfy"
        >
          <svg
            data-testid="check-circle-svg"
            fill="currentColor"
            stroke="none"
          />
        </div>
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`StyledInput shows CircleXSVG when invalid and shouldValidate is true 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 StyledInput__StyledInputWrapper-sc-fkf055-0 escqfF ckFvxY fjNDtr cyFDIt"
      >
        <label
          class="StyledInput__StyledInputLabel-sc-fkf055-2 iHXgrc"
        >
          <input
            class="StyledInput__StyledInputInput-sc-fkf055-1 kkxInS"
            data-testid="styled-input"
            maxlength="255"
            value="abc"
          />
          <div
            class="CombinedStyling__Div-sc-c3jdxk-0 StyledInput__ValidationIcon-sc-fkf055-3 hNEsIT kYVJfy"
          >
            <svg
              data-testid="circle-x-svg"
              fill="currentColor"
              stroke="none"
            />
          </div>
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 StyledInput__StyledInputWrapper-sc-fkf055-0 escqfF ckFvxY fjNDtr cyFDIt"
    >
      <label
        class="StyledInput__StyledInputLabel-sc-fkf055-2 iHXgrc"
      >
        <input
          class="StyledInput__StyledInputInput-sc-fkf055-1 kkxInS"
          data-testid="styled-input"
          maxlength="255"
          value="abc"
        />
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 StyledInput__ValidationIcon-sc-fkf055-3 hNEsIT kYVJfy"
        >
          <svg
            data-testid="circle-x-svg"
            fill="currentColor"
            stroke="none"
          />
        </div>
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
